radix-engine/src/engine/modules/auth_module.rs

```diff
@@ -19,16 +19,47 @@ impl AuthModule {
         method_auths: Vec<MethodAuthorization>,
         call_frames: &Vec<CallFrame>, // TODO remove this once heap is implemented
     ) -> Result<(), RuntimeError> {
-        let mut auth_zones = vec![
-            &call_frames
-                .last()
-                .expect("Current call frame does not exist")
-                .auth_zone,
-        ];
+        let cur_call_frame = call_frames
+            .last()
+            .expect("Current call frame does not exist");
+
+        let auth_zone = cur_call_frame
+            .owned_heap_nodes
+            .values()
+            .find(|e| {
+                matches!(
+                    e,
+                    HeapRootRENode {
+                        root: HeapRENode::AuthZone(..),
+                        ..
+                    }
+                )
+            })
+            .expect("Could not find auth zone")
```

Comment:
I think we should create an `AuthZone` even if the receiver is `&AuthZone` to have a simplified invariant - every frame has a call frame. :)

Reply:
sure :)